### PR TITLE
CMake: Update pybind11 to 2.6.1

### PR DESCRIPTION
**Pull request by knzivid**
2.6.1 is the latest version on conan. The changelog for pybind11
is at https://pybind11.readthedocs.io/en/stable/changelog.html

MISC

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -7,12 +7,16 @@ if(NOT EXISTS "${CMAKE_BINARY_DIR}/conan.cmake")
...
 endif()
 
+find_package(Python3 REQUIRED COMPONENTS Interpreter Development.Module)
 include(${CMAKE_BINARY_DIR}/conan.cmake)
 
-conan_cmake_run(REQUIRES 
-                    eigen/3.3.7@conan/stable
-                    pybind11/2.3.0@conan/stable
-                BASIC_SETUP)
+set(PYBIND11_VERSION 2.6.1)
```
Comment (nedrebo) [inline CMakeLists.txt:13]: Leftover from the previous iteration. This variable is not needed.

File: CMakeLists.txt [hunk 2]
```diff
@@ -7,12 +7,16 @@ if(NOT EXISTS "${CMAKE_BINARY_DIR}/conan.cmake")
                   "${CMAKE_BINARY_DIR}/conan.cmake")
 endif()
 
+find_package(Python3 REQUIRED COMPONENTS Interpreter Development.Module)
```
Comment (nedrebo) [inline CMakeLists.txt:10]: Commit says "Find Python earlier", but not why. Also, this seems to be added and not moved, how did we find Python before?

**Comment by knzivid** [comment]
There are some WIP commits until I get the CI green, bear with me :)

Let's resume the review once green. I can't find the "convert to draft" button

**Comment by knzivid** [comment]
Before I clean up the commits, here are the changes

- Bump the package version in Conan
  - New pybind11 in conan does not export pybind11's CMake targets. It makes its own.
    - conan's pybind11 does not bundle pybind11Config.cmake, so I can't find_package() anymore
    - conan's pybind11 does not import or link with python targets, so we manually do it
- Even in 3.6 tests, cmake can detect 3.9. We pass the python version used for setup.py down to cmake's `find_package(Python3)`

**Comment by knzivid** [comment]
Ready for review, commit cleanup TODO

File: CMakeLists.txt (hunk 2 above)
Comment (knzivid) [reply CMakeLists.txt]: Pybind used to find it for us before. Explained more in the commit message. Let me know if I need to expand more.

**Comment by nedrebo** [comment]
> Before I clean up the commits, here are the changes
> 
> * Bump the package version in Conan
>   
>   * New pybind11 in conan does not export pybind11's CMake targets. It makes its own.
>     
>     * conan's pybind11 does not bundle pybind11Config.cmake, so I can't find_package() anymore
>     * conan's pybind11 does not import or link with python targets, so we manually do it
> * Even in 3.6 tests, cmake can detect 3.9. We pass the python version used for setup.py down to cmake's `find_package(Python3)`

This is bigger changes then when I reviewed the previous version. I'm leaving this review to the maintainers.

**Review by eskaur — APPROVED** [review]

**Pull request merged**